providers/vercel/api.go

```diff
@@ -0,0 +1,170 @@
+// It embeds vercelClient.DNSRecord to reuse the upstream type,
+// but adds fields to handle API inconsistencies (type vs recordType, mxPriority).
+type DNSRecord struct {
+	vercelClient.DNSRecord
+	Type string `json:"type"`
+	// Normally MXPriority would be uint16 type, but since vercelClient.DNSRecord uses int64, we'd better be consistent here
+	// Later in GetZoneRecords we do a `uint16OrZero` to ensure the type is correct
+	MXPriority int64 `json:"mxPriority"`
+}
+
+// pagination represents the pagination object in Vercel API responses.
+type pagination struct {
+	Count int64  `json:"count"`
+	Next  *int64 `json:"next"`
+	Prev  *int64 `json:"prev"`
+}
+
+// listResponse represents the response from the Vercel List DNS Records API.
+type listResponse struct {
+	Records    []DNSRecord `json:"records"`
+	Pagination pagination  `json:"pagination"`
+}
+
+// Vercel API limit is max 100
+const vercelAPIPaginationLimit = 100
+
+// ListDNSRecords retrieves all DNS records for a domain, handling pagination.
+func (c *vercelProvider) ListDNSRecords(ctx context.Context, domain string) ([]DNSRecord, error) {
+	var allRecords []DNSRecord
+	var nextTimestamp int64
+
+	for {
+		url := fmt.Sprintf("https://api.vercel.com/v4/domains/%s/records?limit=%d", domain, vercelAPIPaginationLimit)
+		if c.teamID != "" {
+			url += fmt.Sprintf("&teamId=%s", c.teamID)
+		}
+		if nextTimestamp != 0 {
+			url += fmt.Sprintf("&until=%d", nextTimestamp)
+		}
+
+		var result listResponse
+		err := c.doRequest(clientRequest{
+			ctx:    ctx,
+			method: "GET",
+			url:    url,
+		}, &result, c.listLimiter)
+
+		if err != nil {
+			return nil, fmt.Errorf("failed to list DNS records: %w", err)
+		}
+
+		for _, r := range result.Records {
+			// The official SDK expects 'recordType' but the API returns 'type'.
+			// We explicitly map it here to fix the discrepancy.
+			r.RecordType = r.Type
+			// Ensure Domain field is set (it might not be in the record object itself)
+			if r.Domain == "" {
+				r.Domain = domain
+			}
+			if r.TeamID == "" {
+				r.TeamID = c.teamID
+			}
+
+			allRecords = append(allRecords, r)
+		}
+
+		if result.Pagination.Next == nil {
+			break
+		}
+		nextTimestamp = *result.Pagination.Next
+	}
+
+	return allRecords, nil
+}
+
+// httpsRecord structure for Vercel API
+type httpsRecord struct {
+	Priority int64  `json:"priority"`
+	Target   string `json:"target"`
+	Params   string `json:"params,omitempty"`
+}
+
+// createDNSRecordRequest embeds the official SDK request but adds HTTPS support
+type createDNSRecordRequest struct {
+	vercelClient.CreateDNSRecordRequest
+	HTTPS *httpsRecord `json:"https,omitempty"`
+}
+
+// CreateDNSRecord creates a DNS record.
+func (c *vercelProvider) CreateDNSRecord(ctx context.Context, req createDNSRecordRequest) (*vercelClient.DNSRecord, error) {
+	url := fmt.Sprintf("https://api.vercel.com/v4/domains/%s/records", req.Domain)
+	if c.teamID != "" {
+		url += fmt.Sprintf("?teamId=%s", c.teamID)
```

Comment:
```suggestion
		url += "?teamId=" + c.teamID
```

fmt.Sprintf has extra overhead when running, direct string concat is recommended.